bears/go/GoVetBear.py

```diff
@@ -25,5 +25,130 @@ class GoVetBear:
+            UnreadRune, WriteByte, WriteTo
+        :param nil_function:
+            Comparisons between functions and nil.
+        :param printf:
+            Suspicious calls to functions in the Printf family.
+        :param range_loops:
+            Incorrect uses of range loop variables in closures.
+        :param shadow_variables:
+            Variables that may have been unintentionally shadowed.
+        :param shifts:
+            Shifts equal to or longer than the variable's length.
+        :param struct_tags:
+            Struct tags that do not follow the format understood by
+            reflect.StructTag.Get. Well-known encoding struct tags (json, xml)
+            used with unexported fields.
+        :param tests_and_docs:
+            Mistakes involving tests including functions with incorrect names
+            or signatures and example tests that document identifiers not in
+            the package.
+        :param unreachable:
+            Unreachable code.
+        :param unsafe_pointers:
+            Likely incorrect uses of unsafe.Pointer to convert integers to
+            pointers.
+        :param unused_result:
+            Calls to well-known functions and methods that return a value that
+            is discarded
+        """
+        args = ()
+        # Every check in Vet is true by default.
+        # Adding a flag set to True enables that check and disables all other
+        # checks.
+        # Adding a flag set to False disables that check and runs all
+        # other checks.
+        if assembly_declarations:
+            args += ('-asmdecl=true',)
+        if assign:
+            args += ('-assign=true',)
+        if atomic:
+            args += ('-atomic=true',)
+        if boolean:
+            args += ('-bool=true',)
+        if build_tags:
+            args += ('-buildtags=true',)
+        if invalid_cgo:
+            args += ('-cgocall=true',)
+        if unkeyed_composite_literals:
+            args += ('-composites=true',)
+        if copying_locks:
+            args += ('-copylocks=true',)
+        if http_response:
+            args += ('-httpresponse=true',)
+        if lost_cancel:
+            args += ('-lostcancel=true',)
+        if methods:
+            args += ('-methods=true',)
+        if nil_function:
+            args += ('-nilfunc=true',)
+        if printf:
+            args += ('-printf=true',)
+        if range_loops:
+            args += ('-rangeloops=true',)
+        if shadow_variables:
+            args += ('-shadow=true',)
+        if shifts:
+            args += ('-shift=true',)
+        if struct_tags:
+            args += ('-structtags=true',)
+        if tests_and_docs:
+            args += ('-tests=true',)
+        if unreachable:
+            args += ('-unreachable=true',)
+        if unsafe_pointers:
+            args += ('-unsafeptr=true',)
+        if unused_result:
+            args += ('-unusedresult=true',)
```

Comment:
You should use a list and not tuples, this is way faster with that huge amount of args ;)

Reply:
and I recommend using `itertools.compress`, this compresses the code a bit :) But optional